Providers/Scripts/2.4x-2.5x/Scripts/nxOMSContainers.py

```diff
@@ -0,0 +1,276 @@
+CONF_DIR = os.path.join(MODULE_RESOURCE_DIR, "conf")
+REG_DIR = os.path.join(MODULE_RESOURCE_DIR, "reg")
+LIB_DIR = os.path.join(MODULE_RESOURCE_DIR, "lib")
+YAML_DIR = os.path.join(MODULE_RESOURCE_DIR, "yaml")
+
+
+class IOMSAgent:
+    def restart_oms_agent(self):
+        pass
+    
+class OMSAgentUtil(IOMSAgent):
+    def restart_oms_agent(self, WorkspaceID):
+        wsId = WorkspaceID
+        if wsId is None:
+            wsId = ''
+        if os.system('sudo /opt/microsoft/omsagent/bin/service_control restart ' + wsId) == 0:
+            return True
+        else:
+            LG().Log('ERROR', 'Error restarting omsagent for workspace ' + wsId)
+            return False
+
+class TestOMSAgent(IOMSAgent):
+    def restart_oms_agent(self):
+        return True
+
+OMS_ACTION = OMSAgentUtil()
+
+def init_vars(WorkspaceID = '', Ensure = ''):
+    global CONF_PATH
+    WorkspaceID = WorkspaceID.encode('ascii', 'ignore')
+    Ensure = Ensure.encode('ascii', 'ignore')
+    CONF_PATH = CONF_PATH.replace('<WorkspaceID>',WorkspaceID)
+
+def Set_Marshall(WorkspaceID, Ensure):
+    init_vars(WorkspaceID, Ensure)
+    return Set(WorkspaceID, Ensure)
+
+def Test_Marshall(WorkspaceID, Ensure):
+    init_vars(WorkspaceID, Ensure)
+    return Test(WorkspaceID, Ensure)
+
+def Get_Marshall(WorkspaceID, Ensure):
+    init_vars(WorkspaceID, Ensure)
+
+    retval = 0
+    retd = dict()
+    retd['WorkspaceID'] = WorkspaceID
+    retd['Ensure'] = Ensure
+    return retval, retd
+ 
+def Set(WorkspaceID, Ensure):
+    if Ensure == 'Present':
+        # copy all the required files to the client machine
+        copy_all_files(CONF_DIR, CONF_PATH)
+        update_state_dir_path(os.path.join(CONF_PATH, "container.conf"), WorkspaceID)
+        copy_all_files(PLUGIN_DIR, PLUGIN_PATH)
+        copy_all_files(LIB_DIR, LIB_PATH)
+        copy_all_files(REG_DIR, REG_PATH)
+        check_and_create_yaml(YAML_DIR, YAML_PATH)
+    elif Ensure == 'Absent':
+        # and delete CONF file in the directory
+        delete_all_files(CONF_DIR, CONF_PATH)
+    else:
+        # log error Ensure value not expected
+        LG().Log('ERROR', "Ensure value: " + Ensure + " not expected")
+        return [-1]
+    
+    # restart oms agent
+    if OMS_ACTION.restart_oms_agent(WorkspaceID):
+        return [0]
+    else:
+        return [-1]
+
+def Test(WorkspaceID, Ensure):
+    """
+    Test method for the DSC resoruce
+    If it returns [0] no further action is taken
+    If it returns [-1] Set_Marshall is called
+    """
+
+    # test for the existence of plugin and conf subfolders in the current plugin
+    if Ensure == 'Present':
+        # check all files exist
+        if (not check_all_files(PLUGIN_DIR, PLUGIN_PATH) or not check_all_files(CONF_DIR, CONF_PATH) or not check_all_files(LIB_DIR, LIB_PATH) or not check_all_files(REG_DIR, REG_PATH) or not check_all_files(YAML_DIR, YAML_PATH)):
+            return [-1]
+    elif Ensure == 'Absent':
```

Comment:
Same comment as before: In this case, shouldn't you check that the relevant configuration files are not present? In the Set_Marshall case, you remove them, so you should check here if that needs to be called.
Can you confirm that you installed the resource with Ensure = Present, then installed it with Ensure = Absent and verified that the configuration files were removed?

Reply:
Added the logic to check if the file presence.

Yes, I have tested both the scenarios by installing the resource with Ensure=Present and Ensure=Absent and verified that the conf files were removed.